adapter: properly pass timestamp in ns, not ms

andrioni opened the pull request:
`pack_audit_log_update` was accidentally passing milliseconds instead of nanoseconds when building a timestamp.

### Motivation
  * This PR fixes a previously unreported bug: `pack_audit_log_update` was accidentally passing milliseconds instead of nanoseconds when building a timestamp, which meant the timestamps in the audit log would be wrong.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

mjibson reviewed (commented):
Thanks. This clearly needs a test to ensure we don't make the same mistake again. At least the on disk data are ok, it's just the system table that's wrong, and would be fixed on new envd restart, so we don't need a migration. I'm happy to write a test and push it here.

andrioni commented:
Sounds good to me, but I can also do it if you prefer that way!

I also thought about maybe adding a convenience function to convert from millis to `NaiveDateTime` (since it does happen elsewhere), although I'm not exactly sure where to put that. `ore`?

andrioni commented:
Ah, `ore::now` already had a very similar function, just missing the conversion to `NaiveDateTime` (which is pretty simple), so I switched to using it instead, and added a couple of tests for it just to ensure it is fine. Testing `pack_audit_log_update` seemed a bit too complicated for now.

mjibson reviewed (approved):
Ooh so much better. Thanks for fixing!
